zebra-chain/src/sapling/arbitrary.rs

```diff
@@ -99,3 +102,43 @@ impl Arbitrary for OutputInTransactionV4 {
 
     type Strategy = BoxedStrategy<Self>;
 }
+
+/// A wrapper for a RedJubJub spending verification key.
+///
+/// This is used by proptests since we can't implement Arbitrary for the
+/// redjubjub::VerificationKeyBytes (it's in another crate).
+/// We then implement it for the wrapper type instead.
+#[derive(Debug)]
+struct SpendVerificationKeyBytesWrapper(redjubjub::VerificationKeyBytes<redjubjub::SpendAuth>);
+
+impl Arbitrary for SpendVerificationKeyBytesWrapper {
+    type Parameters = ();
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        (prop::array::uniform32(any::<u8>()))
+            .prop_map(|bytes| {
+                let mut rng = ChaChaRng::from_seed(bytes);
+                let sk = redjubjub::SigningKey::<redjubjub::SpendAuth>::new(&mut rng);
+                let pk = redjubjub::VerificationKey::<redjubjub::SpendAuth>::from(&sk);
+                SpendVerificationKeyBytesWrapper(pk.into())
+            })
+            .boxed()
+    }
+
+    type Strategy = BoxedStrategy<Self>;
+}
```

Comment:
Instead of creating a wrapper type just for proptesting, we can create a `proptest::Strategy` with a function: 

```suggestion
fn spendauth_verification_key_bytes() -> impl Strategy<Value = redjubjub::VerificationKeyBytes<redjubjub::SpendAuth>> {
    prop::array::uniform32(any::<u8>())
        .prop_map(|bytes| {
            let mut rng = ChaChaRng::from_seed(bytes);
            let sk = redjubjub::SigningKey::<redjubjub::SpendAuth>::new(&mut rng);
            redjubjub::VerificationKey::<redjubjub::SpendAuth>::from(&sk).into()
        })
}
```

See note above about invoking it

Reply:
Ah I figured there was a simple way of doing this but wasn't able to come up with it. Thanks! Done in 29b13a0